src/core/core.tooltip.js

```diff
@@ -772,4 +740,79 @@ module.exports = function(Chart) {
 			}
 		}
 	});
+
+	/**
+	 * @namespace Chart.Tooltip.modes
+	 */
+	Chart.Tooltip.modes = {
```

Comment:
I would choose another name because it can be confusing with the `tooltip.mode` option (not sure what name to pick then, maybe `Chart.Tooltip.positioners`).